src/NuGet.Core/NuGet.Frameworks/FrameworkConstants.cs

```diff
@@ -68,6 +68,8 @@ public static class FrameworkIdentifiers
             public const string XamarinXboxOne = "Xamarin.XboxOne";
             public const string UAP = "UAP";
             public const string Tizen = "Tizen";
+            public const string WebAssembly = "WebAssembly";
```

Comment:
Is this `WebAssembly` TFM really appropriate? What really defines the compatibility level between WebAssembly and .NET Standard 2.0? Said another way, is there an official compatibility matrix between WebAssembly and .NET Standard assemblies? Truth be told, I am pretty ignorant about .NET + WebAssembly (and WebAssembly in general). This is the closest thing I could find: https://github.com/xamarin/WebSharp.

At first glance, this seems overly generic and perhaps could be renamed to a the tool that is interopping between .NET Standard assemblies and WebAssembly.

Reply:
To be clear I am not totally pushing back on the name, just trying to understand the full picture.

Reply:
It's [a new framework based on the Mono runtime](http://www.mono-project.com/news/2017/08/09/hello-webassembly/) that runs in the browser and has access to browser APIs. See also https://github.com/mono/mono/tree/master/sdks/wasm. This is the framework used by [Blazor](https://github.com/aspnet/Blazor).

netstandard compat level is the same as the Xamarin frameworks, i.e. unversioned and assumed to be latest.

cc @migueldeicaza @terrajobst

Reply:
The WebAssembly target is compatible with .NET Standard 2, but it is a superset of that API, as it is provides many capabilities not found on .NET Standard 2, it is based on the various mobile profiles used by Xamarin.

I do not think that we need to brand this as a Mono web assembly, it is not an implementation specific profile.

Reply:
@migueldeicaza @mhutch can you please help alleviate @terrajobst 's concern before we merge this?

Reply:
If this TFM was called `MonoWebAssembly`, and then in the future other WebAssembly implementations appear that share some commonality, would it be possible to introduce an intermediate non-implementation-specific `WebAssembly` TFM at that time (so the `MonoWebAssembly` and `OtherWebAssembly` would both become children of `WebAssembly`)?

Reply:
A `WebAssembly` TFM would make sense if there were a stable subset of APIs specific to WebAssembly, a `WebAssemblyStandard 1.0`, such as this:

https://github.com/mono/mono/blob/master/sdks/wasm/bindings.cs#L21

But are we already in such a state to know about this ? Currently, I'm blocked in many occasions by the fact that making nuget packages is not possible, and also because I cannot reference a package that provides this Runtime class.

Similarly, should there be a `WebAssemblyApp` and `MonoWebAssemblyApp` ?

Even if there is only `MonoWebAssembly` or `NetCoreWebAssembly` to begin with, re-targetting to `WebAssembly` when it will make sense is now very easy, and could be dealt with the _runtime_ nuget feature.

Reply:
I agree with @jeromelaban that an abstract `WebAssembly` TFM would like .NET Standard. Which also explains why I don't think we should have another one -- we started .NET Standard specifically to unify all existing portability things (Silverlight, PCL, contracts) into a set that was simpler & viable (i.e. has enough APIs). Moving forward, we don't want specific TFM mappings anymore; an asset is compatible with the project's TFM if it's the same TFM or .NET Standard.

Personally, I think we need a TFM that designates a particular implementation of web assembly, rather than the concept. And `MonoWebAssembly` sounds like a fine plan to me.

Reply:
while MonoWebAssembly would be fine in practice, i want to note that as a library author the reasons i need a tfm here are not mono-specific. i need to make multitargeted builds which do *not* access certain features of .net standard (threading, filesystem) and instead use less efficient implementations or provide fewer features. these are limitations of wasm conceptually, not limitations of mono’s wasm implementation.

this doesn’t matter much since a hypothetical future with `<TargetFrameworks>monowasm;corewasm<TargetFrameworks>` would not be very costly

Reply:
> i need to make multitargeted builds which do not access certain features of .net standard (threading, filesystem) and instead use less efficient implementations or provide fewer features
> these are limitations of wasm conceptually, not limitations of mono’s wasm implementation

I believe these are more the limitations of the browser then of WebAssembly itself. For example, I believe you could write a different WebAssembly host (perhaps something like [wasmjit](https://github.com/rianhunter/wasmjit)) that doesn't have these restrictions.

Reply:
yes, i suppose it's a matter of what 'syscalls' (or DOM bindings) are exposed to the WASM host